h/views/account_signup.py

```diff
@@ -120,29 +121,55 @@ class IDInfo:
     sub: str
 
 
+@dataclass
+class SocialLoginSignupViewsSettings:
+    """Per-route settings for SocialLoginSignupViews."""
+
+    provider: IdentityProvider
+    issuer: JWTIssuers
+    audience: JWTAudiences
```

Comment:
This is the pattern I mentioned in the PR description, which you'll see repeated a few times in the views below: this dataclass contains all the provider-specific values needed by the `SocialLoginSignupViews` class. One object-instance of this settings class will be constructed for each provider and the view will select the correct settings class based on the route being currently handled (there are separate routes for each provider).